src/components/TextField/TextField.tsx

```diff
@@ -104,18 +179,41 @@ export class TextField extends React.Component<TextFieldProps> {
           className,
         )}
         ref={this.TextFieldRef}
-        suffix={
-          suffixToShow && (
-            <div className={classes.suffixWrapper}>
-              <div className={classes.gap} />
-              {suffixToShow}
-            </div>
-          )
-        }
+        suffix={this._getSuffix()}
         error={error}
         {...restProps}
         disabled={disabled}
       />
     );
   }
 }
+
+const ErrorSuffix = ({ errorMessage }) => (
+  <Tooltip
+    appendTo="scrollParent"
+    placement="top-end"
+    skin={TooltipSkin.Error}
+    content={errorMessage}
+    moveBy={{ x: 5, y: 0 }}
+  >
+    <ErrorIcon />
+  </Tooltip>
+);
+
+const StatusIcon = ({
+  className,
+  error,
+  errorMessage,
+  success,
+  successIcon,
+}) => {
+  let statusIcon = null;
+
+  if (errorMessage && error) {
+    statusIcon = <ErrorSuffix errorMessage={errorMessage} />;
+  } else if (successIcon && success) {
+    statusIcon = <SuccessIcon data-hook={DATA_HOOKS.SUCCESS_ICON} />;
+  }
```

Comment:
Note that in the [original implementation](https://github.com/wix/wix-ui-tpa/blob/master/src/components/TextField/TextField.tsx#L65), the `success` status had precedent, cause it didn't have the `else`.

So it should either be flipped:
```tsx
   if (successIcon && success) {
    statusIcon = <SuccessIcon data-hook={DATA_HOOKS.SUCCESS_ICON} />;
  } else if (errorMessage && error) {
    statusIcon = <ErrorSuffix errorMessage={errorMessage} />;
  }
```
or:
```tsx
  if (errorMessage && error) {
    statusIcon = <ErrorSuffix errorMessage={errorMessage} />;
  }
  if (successIcon && success) {
    statusIcon = <SuccessIcon data-hook={DATA_HOOKS.SUCCESS_ICON} />;
  }
```

Reply:
I know that it was the opposite but there was a bug: when you passed all of the props I mean on this case: <TextField successIcon success error errorMessage /> the border of the input was colored red (error border) but the suffix icon was a success icon.
So I fixed it:)
![Screen Shot 2020-11-04 at 10 06 36](https://user-images.githubusercontent.com/66675260/98085138-8d988a00-1e85-11eb-8113-ca28c55bd3fb.png)